File: corpus/rules/vibe-secret-in-log/fp/src/safe.ts
// FP fixture for vibe-secret-in-log.

declare const logger: { info: (...a: unknown[]) => void };
declare function redact<T>(v: T): T;
declare function mask(s: string): string;
declare function sanitize<T>(v: T): T;

export function logSafeStrings() {
  console.log('hello world');                         // no secret-name
  console.error('error', { user: 'alice' });          // no secret-name
  logger.info({ count: 42, name: 'demo' });           // benign keys
}

export function logRedacted(apiKey: string, accessToken: string) {
  console.log({ apiKey: redact(apiKey) });            // redacted — silent
  logger.info('token=' + mask(accessToken));          // mask() wrapping
  console.warn(sanitize({ accessToken }));            // sanitize call wraps
}

// Non-logger calls that mention secret names are NOT flagged.
export function useApiKey(apiKey: string): string {
  return `Bearer ${apiKey}`;                          // no logger involved
}

File: corpus/rules/vibe-py-secret-in-log/fp/src/safe.py
import logging

logger = logging.getLogger(__name__)


def redact(v): return '***'
def mask(v): return '***'


def log_safe():
    logger.info('startup complete')                  # no secret-shaped name
    logger.info({'count': 42, 'name': 'demo'})       # benign dict keys
    logger.warning('error', extra={'request_id': 'abc'})


def log_redacted(api_key, access_token):
    logger.info({'apiKey': redact(api_key)})         # wrapped — silent
    logger.error('token=' + mask(access_token))      # wrapped — silent


# Non-logger functions with secret-named args are NOT flagged.
def use_api_key(api_key):
    return f'Bearer {api_key}'
